src/cutadapt/script.sh

```diff
@@ -0,0 +1,177 @@
+  ${par_cut:+--cut "${par_cut}"} \
+  ${par_cutR2:+--cutR2 "${par_cutR2}"} \
+  ${par_nextseq_trim:+--nextseq-trim "${par_nextseq_trim}"} \
+  ${par_quality_cutoff:+--quality-cutoff "${par_quality_cutoff}"} \
+  ${par_quality_cutoffR2:+--quality-cutoffR2 "${par_quality_cutoffR2}"} \
+  ${par_quality_base:+--quality-base "${par_quality_base}"} \
+  ${par_poly_a:+--poly-a} \
+  ${par_length:+--length "${par_length}"} \
+  ${par_trim_n:+--trim-n} \
+  ${par_length_tag:+--length-tag "${par_length_tag}"} \
+  ${par_strip_suffix:+--strip-suffix "${par_strip_suffix}"} \
+  ${par_prefix:+--prefix "${par_prefix}"} \
+  ${par_suffix:+--suffix "${par_suffix}"} \
+  ${par_rename:+--rename "${par_rename}"} \
+  ${par_zero_cap:+--zero-cap} \
+)
+echo "Arguments to cutadapt:"
+echo $mod_args
+echo
+
+# Filtering of processed reads arguments
+###########################################################
+echo ">> Filtering of processed reads arguments"
+[[ "$par_discard_trimmed" == "false" ]] && unset par_discard_trimmed
+[[ "$par_discard_untrimmed" == "false" ]] && unset par_discard_untrimmed
+[[ "$par_discard_casava" == "false" ]] && unset par_discard_casava
+
+filter_args=$(echo \
+  ${par_minimum_length:+--minimum-length "${par_minimum_length}"} \
+  ${par_maximum_length:+--maximum-length "${par_maximum_length}"} \
+  ${par_max_n:+--max-n "${par_max_n}"} \
+  ${par_max_expected_errors:+--max-expected-errors "${par_max_expected_errors}"} \
+  ${par_max_average_error_rate:+--max-average-error-rate "${par_max_average_error_rate}"} \
+  ${par_discard_trimmed:+--discard-trimmed} \
+  ${par_discard_untrimmed:+--discard-untrimmed} \
+  ${par_discard_casava:+--discard-casava} \
+)
+echo "Arguments to cutadapt:"
+echo $filter_args
+echo
+
+# Output arguments
+# We write the output to a directory rather than
+# individual files.
+###########################################################
+echo ">> Output arguments"
+[[ "$par_json" == "false" ]] && unset par_json
+[[ "$par_fasta" == "false" ]] && unset par_fasta
+[[ "$par_info_file" == "false" ]] && unset par_info_file
+
+if [ $mode = "se" ]; then
+  output_args=$(echo \
+    ${par_report:+--report "${par_report}"} \
+    ${par_json:+--json "${par_output}/report.json"} \
+    --output "$par_output/{name}_R1_001.fastq" \
+    ${par_fasta:+--fasta} \
+    ${par_info_file:+--info-file} \
+  )
+else
+  output_args=$(echo \
+    ${par_report:+--report "${par_report}"} \
+    ${par_json:+--json "${par_output}/report.json"} \
+    --output "$par_output/{name}_R1_001.fastq" \
+    --paired-output "$par_output/{name}_R2_001.fastq" \
+    ${par_fasta:+--fasta} \
+    ${par_info_file:+--info-file} \
+  )
+fi
+echo "Arguments to cutadapt:"
+echo $output_args
+echo
+
+echo ">> Full CLI to be run:"
+cli=$(echo \
+  $input \
+  $adapter_args \
+  $paired_args \
+  $input_args \
+  $mod_args \
+  $filter_args \
+  $output_args
+)
+
+echo cutadapt $cli | sed -e 's/--/\r\n  --/g'
+
+cutadapt $cli | tee $par_output/report.txt
```

Comment:
```suggestion
cutadapt $cli
```

IMO the user should tee if they want to retain the cutadapt STDOUT, no?

Reply:
The report is effectively written to stdout and (according to the people I know use this) is crucial as an output. In another pipeline we only redirect it to the report file but I prefer to see the output in the process log as well.

Reply:
Ok, fair enough!